src/cmd/partition.rs

```diff
@@ -0,0 +1,223 @@
+    --filename <filename>  A filename template to use when constructing
+                           the names of the output files.  The string '{}'
+                           will be replaced by a value based on the value
+                           of the field, but sanitized for shell safety.
+                           [default: {}.csv]
+
+Common options:
+    -h, --help             Display this message
+    -o, --output <file>    Write output to <file> instead of stdout.
+    -n, --no-headers       When set, the first row will NOT be interpreted
+                           as column names. Otherwise, the first row will
+                           appear in all chunks as the header row.
+    -d, --delimiter <arg>  The field delimiter for reading CSV data.
+                           Must be a single character. (default: ,)
+";
+
+#[derive(Clone, RustcDecodable)]
+struct Args {
+    arg_column: SelectColumns,
+    arg_input: Option<String>,
+    arg_outdir: String,
+    flag_filename: FilenameTemplate,
+    // This option is unused, both here and in the split command.
+    flag_output: Option<String>,
+    flag_no_headers: bool,
+    flag_delimiter: Option<Delimiter>,
+}
+
+pub fn run(argv: &[&str]) -> CliResult<()> {
+    let args: Args = try!(util::get_args(USAGE, argv));
+    try!(fs::create_dir_all(&args.arg_outdir));
+
+    // It would be nice to support efficient parallel partitions, but doing
+    // do would involve more complicated inter-thread communication, with
+    // multiple readers and writers, and some way of passing buffers
+    // between them.
+    args.sequential_partition()
+}
+
+impl Args {
+    /// Configuration for our reader.
+    fn rconfig(&self) -> Config {
+        Config::new(&self.arg_input)
+            .delimiter(self.flag_delimiter)
+            .no_headers(self.flag_no_headers)
+            .select(self.arg_column.clone())
+    }
+
+    /// Get the column to use as a key.
+    fn key_column(&self, rconfig: &Config, headers: &[csv::ByteString])
+                  -> CliResult<usize> {
+        let select_cols = try!(rconfig.selection(&*headers));
+        if select_cols.len() == 1 {
+            Ok(select_cols[0])
+        } else {
+            Err("can only partition on one column".to_owned().into())
+        }
+    }
+
+    /// A basic sequential partition.
+    fn sequential_partition(&self) -> CliResult<()> {
+        let rconfig = self.rconfig();
+        let mut rdr = try!(rconfig.reader());
+        let headers = try!(rdr.byte_headers());
+        let key_col = try!(self.key_column(&rconfig, &*headers));
+        let mut gen = WriterGenerator::new(self.flag_filename.clone());
+
+        let mut writers: HashMap<csv::ByteString, BoxedWriter> =
+            HashMap::new();
+        for row in rdr.byte_records() {
+            let row = try!(row);
+
+            // Decide what file to put this in.
+            let key = row[key_col].clone();
+            let mut entry = writers.entry(key.clone());
```

Comment:
No, you need both as far as I can tell. We call `try!(wtr.write(row.into_iter()));` below, which consumes the underlying backing `row`, and which leads to borrow checker problems. The easiest workaround it to make our own copy of `key`. I also tried something clever with `let ref key` and `drop(key)`, but it wouldn't work without non-lexical lifetimes.